compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -972,6 +972,27 @@ public BSymbol lookupMemberSymbol(Location pos,
         return symTable.notFoundSymbol;
     }
 
+    /**
+     * Return the symbol with the given name even with a different package.
+     *
+     * @param scope     current scope
+     * @param name      symbol name
+     * @param expSymTag expected symbol type/tag
+     * @return resolved symbol
+     */
+    public BSymbol lookupPossibleMemberSymbol(Scope scope, Name name, long expSymTag) {
```

Comment:
The function `lookupMemberSymbol` works similar to this function. What we have to do is create a function that looks for member symbol (keep lookupMemberSymbol function). Then, wherever we need to check if access to that symbol is allowed, we need to do `isMemberAccessAllowed` check